Add HPPS instructions

m1r0 opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/7415

## Proposed Changes

This PR adds HPPS instructions to the settings screen. It also adds a link to the future docs page.

![jsWU9w.png](https://github.com/Automattic/sensei/assets/1612178/577fa7c6-2d62-4be3-99a2-a995179c9ac3)

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Go to Sensei LMS -> Settings -> Experimental Features
2. Enable the HPPS feature.
3. You should see the instructions.

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [x] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [x] Hooks (p6rkRX-1uS-p2) and functions are documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [x] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [x] Legacy courses (course without blocks) are tested
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [x] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [x] "Needs Documentation" label is added if this change requires updates to documentation
- [x] Known issues are created as new GitHub issues

m1r0 committed:
Add HPPS instructions

m1r0 committed:
Add changelog

donnapep reviewed (commented):
Looking a lot better!

Given this is currently our only experimental feature, WDYT about making those instructions always visible? Currently, as soon as you check the box, step 1 is already technically fulfilled so people may wonder "Well, didn't I just do that?" And I think there may be a step missing to enable the sync option?

I think we could also make the steps even more explicit. For example, something like:

1. Select the "High-Performance Progress Storage" checkbox.
2. Save the changes.
3. Select the "Progress storage synchronization" checkbox.
4. Save the changes.
5. Wait until the "Migration complete and data synchronization enabled" message is displayed. This may take awhile and you will need to refresh the page to see the updated status.
6. Select the "High-Performance progress storage (experimental)" option."
7. Save the changes.
8. You are now using High-Performance Progress Storage!

The save steps could also be part of the previous step if we want to make it more concise.

Could we also reorder the steps so that they match the order the user should enable them in? So "Progress storage synchronization" would go before "Progress storage repository".

donnapep commented:
Oops, didn't mean to approve. 😅

m1r0 committed:
Update the HPPS instructions

m1r0 commented:
Thanks for the input, @donnapep!

> Given this is currently our only experimental feature, WDYT about making those instructions always visible?

Yep, makes sense. I've updated the instructions in ee779b679ef5af2d0c235bb1160989d0ec326e6c. Here's a quick screenshot.

![xz0iaL.png](https://github.com/Automattic/sensei/assets/1612178/15ffe6f2-d7d0-4382-9e71-46121bcaaa89)

I've also added an extra step for actually enabling the feature but let me know if that's too much.

The "save the changes" part looks a bit repetitive and kinda obvious. Should we leave it in? 

> Currently, as soon as you check the box, step 1 is already technically fulfilled so people may wonder "Well, didn't I just do that?" And I think there may be a step missing to enable the sync option?

I'm guessing you've misread the first step in the initial instuctions. It's for enabling the sync. 🙂

donnapep commented:
> The "save the changes" part looks a bit repetitive and kinda obvious. Should we leave it in?

It is, but I think it's important to mention it since I've noticed myself missing the Save button on occasion. We could trim it down to just "save" though.

I'm wondering about this bit?

> Could we also reorder the steps so that they match the order the user should enable them in? So "Progress storage synchronization" would go before "Progress storage repository".

donnapep reviewed (dismissed):
Dropped a comment https://github.com/Automattic/sensei/pull/7420#issuecomment-1889744174, otherwise 👍🏻 .

m1r0 committed:
Change the order of the HPPS settings

m1r0 commented:
> Could we also reorder the steps so that they match the order the user should enable them in? So "Progress storage synchronization" would go before "Progress storage repository".

Do you mean to change the order of the settings? If that's the case, I've updated it in https://github.com/Automattic/sensei/pull/7420/commits/a62fe37068b03f65354c2391b1fc02b3d595cb5b.
